mettle/api/api.go

```diff
@@ -91,19 +95,59 @@ type server struct {
 	mutex        sync.Mutex
 }
 
-// ServeExecutions serves a list of currently executing jobs over HTTP.
-func (s *server) ServeExecutions(w http.ResponseWriter, r *http.Request) {
-	resp := map[string]string{}
-	m := jsonpb.Marshaler{Indent: "  "}
+// ServeExecutions serves a list of currently executing jobs over GRPC.
+func (s *server) ServeExecutions(ctx context.Context, req *bpb.ServeExecutionsRequest) (*bpb.ServeExecutionsResponse, error) {
+	log.Notice("Received request for inflight executions")
 	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	executions := []*bpb.Job{}
 	for k, v := range s.jobs {
-		s, _ := m.MarshalToString(v.Current)
-		resp[k] = s
+		current, _ := proto.Marshal(v.Current)
+		job := &bpb.Job{
+			ID:        k,
+			Current:   current,
+			SentFirst: v.SentFirst,
+			Done:      v.Done,
+		}
+		executions = append(executions, job)
 	}
-	s.mutex.Unlock()
-	e := json.NewEncoder(w)
-	e.SetIndent("", "  ")
-	e.Encode(resp)
+	res := &bpb.ServeExecutionsResponse{
+		Jobs: executions,
+	}
+	log.Notice("Serving %d inflight executions", len(executions))
+	return res, nil
+}
+
+// getExecutions requests a list of currently executing jobs over grpc and updates the server.
+func getExecutions(opts grpcutil.Opts, conTLS bool) (map[string]*job, error) {
+	conn, err := grpcutil.Dial(fmt.Sprintf("%s:%d", opts.Host, opts.Port), conTLS, opts.CertFile, opts.TokenFile)
+	if err != nil {
+		return map[string]*job{}, fmt.Errorf("Failed to connect to bootstrap client: %s", err)
+	}
+	defer conn.Close()
+	client := bpb.NewBootstrapClient(conn)
+	log.Notice("Requesting inflight executions...")
+	req := &bpb.ServeExecutionsRequest{}
+	res, err := client.ServeExecutions(context.Background(), req)
+	if err != nil {
+		return map[string]*job{}, fmt.Errorf("Failed to get inflight executions: %s", err)
+	}
+	jobs := make(map[string]*job, len(res.Jobs))
+	for _, j := range res.Jobs {
+		current := &longrunning.Operation{}
+		if err := proto.Unmarshal(j.Current, current); err != nil {
+			log.Warningf("unable to unmarshal s%", j.ID)
+			continue
+		}
+		job := &job{
+			Current:   current,
+			SentFirst: j.SentFirst,
+			Done:      j.Done,
+		}
+		jobs[j.ID] = job
```

Comment:
nit: just `jobs[j.ID] = &job{ ... }`, you don't use the local variable for anything